sklearn/utils/_array_api.py

```diff
@@ -735,6 +735,26 @@ def _nanmax(X, axis=None, xp=None):
         return X
 
 
+def _nanmean(X, axis=None, xp=None):
+    # TODO: refactor once nan-aware reductions are standardized:
+    # https://github.com/data-apis/array-api/issues/621
+    xp, _ = get_namespace(X, xp=xp)
+    if _is_numpy_namespace(xp):
+        return xp.asarray(numpy.nanmean(X, axis=axis))
+    else:
+        mask = xp.isnan(X)
+        total = xp.sum(xp.where(mask, xp.asarray(0.0, device=device(X)), X), axis=axis)
+        count = xp.sum(
+            xp.where(
+                mask,
+                xp.asarray(0.0, device=device(X)),
+                xp.asarray(1.0, device=device(X)),
+            ),
+            axis=axis,
+        )
```

Comment:
The following looks more straightforward:


```suggestion
        count = xp.sum(xp.astype(xp.logical_not(mask), X.dtype), axis=axis)
```

Reply:
Nice!